clamrest.go

```diff
@@ -143,24 +192,9 @@ func scanHandler(w http.ResponseWriter, r *http.Request) {
 			fmt.Printf(time.Now().Format(time.RFC3339) + " Started scanning: " + part.FileName() + "\n")
 			var abort chan bool
 			response, err := c.ScanStream(part, abort)
-			for s := range response {
-				w.Header().Set("Content-Type", "application/json; charset=utf-8")
-				respJson := fmt.Sprintf("{ \"Status\": \"%s\", \"Description\": \"%s\" }", s.Status, s.Description)
-				switch s.Status {
-				case clamd.RES_OK:
-					w.WriteHeader(http.StatusOK)
-				case clamd.RES_FOUND:
-					w.WriteHeader(http.StatusNotAcceptable)
-				case clamd.RES_ERROR:
-					w.WriteHeader(http.StatusBadRequest)
-				case clamd.RES_PARSE_ERROR:
-					w.WriteHeader(http.StatusPreconditionFailed)
-				default:
-					w.WriteHeader(http.StatusNotImplemented)
-				}
-				fmt.Fprint(w, respJson)
-				fmt.Printf(time.Now().Format(time.RFC3339)+" Scan result for: %v, %v\n", part.FileName(), s)
-			}
+
+			handleScanResults(response, part.FileName(), w)
```

Comment:
This will falsely return a 200 OK if a FormPost with multiple files are used to call the /scan endpoint and the first file is clean, but the second one contains a virus, once you w.WriteHeader(http.Status), the response status will not change if you write it again. Maybe you are aware of this, considering your comment about limitations. [This PR fixes that bug](https://github.com/ajilach/clamav-rest/pull/50/files)

Reply:
You're right - this is of course a security hole. Hope the maintainer merges your PR... then we could add more endpoints and get rid of some code duplications.

Reply:
Yes, it's good that you're also interested in adding functionality and patching holes, it gives some potential for the projects future. I'm also interested in enhancement and bug fixes.

Reply:
Gentlemen, thanks a lot for your contributions. I did not react until now simply because I did not notice your open PRs. Sorry for that. Will try to set Github alerts for PRs going forward.

As a next step, can you update and test your changes against the current code base (fixed the database update)?